### ISSUE TITLE
Sorting of plugin search results

### ISSUE THREAD

**Issue by jennydaman** [FNNDSC/ChRIS_store_ui#135]
https://github.com/FNNDSC/ChRIS_store_ui/blob/ca45de40784ddb979445252c3d71c05bc3699b50/src/components/Plugins/Plugins.js#L218

Some sorts could be:

- alphabetical, by author
- alphabetical, by name
- chronological, by date created

**Comment by sonylomo** [comment]
Can I work on this one @jennydaman?

**Comment by sonylomo** [comment]
![conflict](https://user-images.githubusercontent.com/49971500/114519936-928a1280-9c49-11eb-88df-e54337479c6c.png)
I'm trying to resolve conflicts but my code isn't passing these 2 tests. I didn't change any code related to them. I've checking what's wrong and updated my branch to be inline with the master before fixing plugin sort.

I've also checked if the  "mark as favorite" functionality on plugins store website is working but it's not - was it working before?

**Comment by jennydaman** [comment]
Try this:

```bash
git remote add upstream git@github.com:FNNDSC/ChRIS_store_ui.git
git fetch upstream
git diff upstream/master
```

I found this:

```patch
diff --git a/src/components/Plugins/Plugins.js b/src/components/Plugins/Plugins.js
index a157f49..e5e2b45 100755
--- a/src/components/Plugins/Plugins.js
+++ b/src/components/Plugins/Plugins.js
@@ -113,20 +114,23 @@ export class Plugins extends Component {
     }
   }

-  fetchPlugins = () => {
-    const params = new URLSearchParams(window.location.search)
-    const name = params.get('q') //get value searched from the URL
+  fetchPlugins() {
+    const params = new URLSearchParams(window.location.search);
+    const name = params.get("q"); //get value searched from the URL
+
     const searchParams = {
       limit: 20,
       offset: 0,
       name_title_category:name,
-    };
+    }
+
```

Notice the declaration of `fetchPlugins` was changed. This is an oddity with syntax. If a function is defined as `fetchPlugins() {`, `this` will be undefined unless you manually "bind" it.

**Comment by sonylomo** [comment]
> Notice the declaration of `fetchPlugins` was changed. This is an oddity with syntax. If a function is defined as `fetchPlugins() {`, `this` will be undefined unless you manually "bind" it.

Thanks for that. What about the `starsByPlugin` test?

**Comment by jennydaman** [comment]
@sonylomo all tests are passing on https://github.com/sonylomo/ChRIS_store_ui/commit/b154b309e04425a781d0a8f59aa033f665388d45

**Comment by sonylomo** [comment]
Yeah, I spotted the problem with the tests. Everything is ok now.

**Comment by sonylomo** [comment]
I've closed the other PR because I deleted that forked repo 😁.
